src/Workspaces/Core/Portable/Extensions/OperationExtensions.cs

```diff
@@ -42,6 +42,29 @@ public static ValueUsageInfo GetValueUsageInfo(this IOperation operation)
             | typeof(x)       |      |       |             |             |       ✔️        | ️
 
             */
+            if (operation is ILocalReferenceOperation localReference &&
+                localReference.IsDeclaration &&
+                !localReference.IsImplicit) // Workaround for https://github.com/dotnet/roslyn/issues/30753
+            {
+                // Declaration expression.
+                return ValueUsageInfo.Write;
+            }
+            else if (operation is IDeclarationPatternOperation)
+            {
+                switch (operation.Parent)
+                {
+                    case IPatternCaseClauseOperation _:
+                        return ValueUsageInfo.Write;
+
+                    case IIsPatternOperation _:
+                        return ValueUsageInfo.ReadWrite;
+
+                    default:
+                        Debug.Fail("Unhandled declaration pattern context");
+                        // Conservatively assume read/write.
+                        return ValueUsageInfo.ReadWrite;
+                }
+            }
```

Comment:
1. needs explanations of 'why'
2. there's a table above, but it doesn't seem to have been updated to handle thse cases.